src/components/Charts/ChataPieChart/ChataPieChart.js

```diff
@@ -0,0 +1,360 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import _get from 'lodash.get'
+import _cloneDeep from 'lodash.clonedeep'
+import { v4 as uuid } from 'uuid'
+import { select } from 'd3-selection'
+import { scaleOrdinal } from 'd3-scale'
+import { pie, arc } from 'd3-shape'
+import { entries } from 'd3-collection'
+import legendColor from '../D3Legend/D3Legend'
+
+import { deepEqual, formatElement, removeFromDOM } from '../../../js/Util'
+import { chartDefaultProps, chartPropTypes, getTooltipContent } from '../helpers'
+import { getChartColorVars } from '../../../theme/configureTheme'
+
+import 'd3-transition'
+
+export default class ChataPieChart extends Component {
+  constructor(props) {
+    super(props)
+
+    this.CHART_ID = uuid()
+    this.LEGEND_ID = `react-autoql-pie-legend-${uuid()}`
+
+    this.sortedData = props.data
+      .concat() // this copies the array so the original isn't mutated
```

Comment:
Nice trick! I think recently the structuredClone() is now supported by all the main browsers, so this might be a good alternative to use.